src/Data/Functor/Plus.hs

```diff
@@ -81,10 +81,18 @@ import GHC.Generics
 -- > m <!> zero = m
 --
 -- If extended to an 'Alternative' then 'zero' should equal 'empty'.
-
 class Alt f => Plus f where
   zero :: f a
 
+-- | The sum of a collection of actions, generalizing 'concat'.
+--
+-- > asum [Just "Hello", Nothing, Just "World"]
+-- >   -- Just "Hello"
```

Comment:
This should be `psum`, I think. Also, we might as well convert this to an actual doctest:

```suggestion
-- >>> psum [Just "Hello", Nothing, Just "World"]
-- Just "Hello"
```